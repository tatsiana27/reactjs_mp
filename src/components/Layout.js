import React from 'react';
import { Container } from 'semantic-ui-react';

import { CommonComponent } from './CommonComponent';
import './layout.scss';

export function Layout({ children }) {
    return (
        <Container>
            {children}
            <CommonComponent/>
        </Container>
    );
};