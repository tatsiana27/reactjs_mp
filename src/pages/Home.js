import React, { Component } from 'react';
import { Layout } from './Layout';
import { Header } from '../components/Header';
import { SearchBlock } from '../components/SearchBlock';
import { InfoBlock } from '../components/InfoBlock';
import { MoviesList } from '../components/MoviesList';

export class Home extends Component {

    render() {
        return (
            <Layout>
                <Header>
                    <SearchBlock/>
                </Header>
                <InfoBlock/>
                <MoviesList/>
            </Layout>
        )
    }
};
