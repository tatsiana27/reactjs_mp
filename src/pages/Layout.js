import React from 'react';
import { Container } from 'semantic-ui-react';

import { CommonComponent } from '../components/CommonComponent';
import '../styles/layout.scss';

export function Layout({ children }) {
    return (
        <Container>
            {children}
            <CommonComponent/>
        </Container>
    );
};