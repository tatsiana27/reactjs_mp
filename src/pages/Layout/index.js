import React from 'react';
import PropTypes from 'prop-types';
import { Container } from 'semantic-ui-react';

import { CommonComponent } from '../../components/CommonComponent';
import './Layout.scss';

export function Layout({ children }) {
  return (
    <Container>
      {children}
      <CommonComponent/>
    </Container>
  );
};

Layout.defaultProps = {
  children: null,
};

Layout.propTypes = {
  children: PropTypes.node,
};
