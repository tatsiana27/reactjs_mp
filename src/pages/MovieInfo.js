import React, { Component } from 'react';
import { Layout } from './Layout';
import { Header } from '../components/Header';
import { Movie } from '../components/Movie';
import { MoviesList } from '../components/MoviesList';

export class MovieInfo extends Component {

    render() {
        const movieId = this.props.match.params.id;
        return (
            <Layout>
                <Header>
                    <Movie id={movieId}/>
                </Header>
                <MoviesList/>
            </Layout>
        )
    }
};