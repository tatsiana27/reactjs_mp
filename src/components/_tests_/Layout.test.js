import React from 'react';
import { Layout } from '../../pages/Layout';
import { shallow } from 'enzyme';

describe('<Layout />', () => {
    test('should render Layout Component', () => {
        const wrapper = shallow(
            <Layout />
        );

        expect(wrapper).toMatchSnapshot();
    });
});