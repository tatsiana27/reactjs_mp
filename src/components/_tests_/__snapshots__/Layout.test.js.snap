// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should render Layout Component 1`] = `
<Container>
  <CommonComponent />
</Container>
`;
